src/openms/source/ANALYSIS/MAPMATCHING/QTClusterFinder.cpp

```diff
@@ -239,52 +241,57 @@ namespace OpenMS
 
     // compute QT clustering:
     // std::cout << "Clustering..." << std::endl;
-    list<QTCluster> clustering;
-    computeClustering_(grid, clustering);
-    // number of clusters == number of data points:
-    Size size = clustering.size();
 
-    // create a temp. map storing which grid features are next to which clusters
-    typedef OpenMSBoost::unordered_map<Size, std::vector<GridFeature*> > NeighborList;
+    // "hot" cluster heads, we can extract the best efficiently 
+    Heap cluster_heads;
+
+    // handles to cluster heads to reach them (index == cluster.id_) in cluster_heads for updating
+    vector<Handle> handles;
+
+    // "cold" cluster bodies, where most of their data lies
+    vector<QTCluster::Data_> cluster_data;
+
+    // map to get ids from clusters, who contain a certain grid feature
     ElementMapping element_mapping;
-    for (list<QTCluster>::iterator it = clustering.begin();
-         it != clustering.end(); ++it)
+
+    computeClustering_(grid, cluster_heads, cluster_data, handles);
+
+    // number of clusters == number of data points:
+    Size size = cluster_heads.size();
+
+    // fill the element mapping
+    for (Heap::iterator it = cluster_heads.begin();
+         it != cluster_heads.end(); ++it)
     {
-      NeighborList neigh = it->getAllNeighbors();
-      for (NeighborList::iterator n_it = neigh.begin(); n_it != neigh.end(); ++n_it)
+      NeighborMap const& neigh = it->getAllNeighborsDirect();
```

Comment:
we use a different style throughout OpenMS:
const NeighborMap&
So changed this and other places would make it more consistent.

Reply:
Done.